lib/train/platforms/detect/specifications/os.rb

```diff
@@ -17,9 +17,17 @@ def self.load
 
       plat.family('windows').in_family('os')
           .detect {
-            if winrm? || (@backend.local? && ruby_host_os(/mswin|mingw32|windows/))
-              true
+            # Can't return from a `proc` thus the `is_windows` shenanigans
+            is_windows = false
+            is_windows = true if winrm?
+
+            if defined?(Train::Transports::Local::Connection)
+              if @backend.class == Train::Transports::Local::Connection
+                is_windows = true
+              end
```

Comment:
This could also be `if winrm? || (@backend.class.to_s == 'Train::Transports::Local::Connection' && ruby_host_os(/mswin|mingw32|windows/))`

Reply:
I thought that, but you have to check if it is defined as well otherwise you get an error.

Reply:
the to_s kinda gets around that but im fine either way

Reply:
Ah! I like that more. I'll do that.